File: 3740-minimum-distance-between-three-equal-elements-i/3740-minimum-distance-between-three-equal-elements-i.py
class Solution(object):
    def minimumDistance(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        dict = {}

        for i in range( 0 , len(nums) ):
            if( dict.get(nums[i])):
                dict[nums[i]].append(i)
            else:
                dict[nums[i]] = [i]

        
        result = 1000000

        for key in dict:
            
            arr = dict[key]
            if( len(arr) > 2 ):
                for j in range( 0 , len(arr) - 2 ):
                    a = arr[j] 
                    b = arr[j+1] 
                    c = arr[j+2]

                    result = min( result , ( c - a ) + ( b - a ) + ( c - b ))
        

        if( result == 1000000 ):
            return -1
        return result 

File: 3740-minimum-distance-between-three-equal-elements-i/3740-minimum-distance-between-three-equal-elements-i.cpp
class Solution {
public:
    int minimumDistance(vector<int>& nums) {
        
        unordered_map<int,vector<int>> MAP ;

        for( int i = 0 ; i < nums.size() ; i++ ){
            MAP[nums[i]].push_back(i) ;
        }

        int result = INT_MAX ;

        for( auto i : MAP ){

            vector<int> temp = i.second ;
            
            if( temp.size() > 2 ){
                for( int j = 0 ; j < temp.size() - 2 ; j++ ){
                    int a = temp[j] , b = temp[j+1] , c = temp[j+2] ;
                    result = min( result , ( b - a ) + ( c - b ) + ( c - a ) );
                }
            }   
        }

        return result == INT_MAX ? -1 : result ;

    }
};